components/feature/downloads/src/main/java/mozilla/components/feature/downloads/manager/FetchDownloadManager.kt

```diff
@@ -76,7 +76,7 @@ class FetchDownloadManager<T : AbstractFetchDownloadService>(
     }
 
     override fun tryAgain(downloadId: Long) {
-        val download = queuedDownloads[downloadId]
+        val download = queuedDownloads[downloadId] ?: return
```

Comment:
Are there any steps to reproduce this issue? I want to find the root cause, If we do that, are we just preventing the crash, but the notification is going to persist, without any behaviour? I noticed in [unregisterListeners](https://github.com/mozilla-mobile/android-components/blob/ba60c8df899b4e6fd32341fc6a2fc011b1009376/components/feature/downloads/src/main/java/mozilla/components/feature/downloads/manager/FetchDownloadManager.kt#L91), we are clearing the `queuedDownloads`, should we also remove all the pending notifications too, as the `downloads` in the downloads notification are not going to be valid anymore?

Reply:
I didn't find any good way to reproduce it.  I can only hard code the failure locally in my testing.  You bring up a good point.  Since `unregisterListeners` clears the download queue we should also clear all download notifications.  I'll look more into this.  Thanks,

Reply:
👍 
We have to be careful if we clear notifications as they could be in a downloading state

Reply:
Sure, I'll confirm if clearing notification is doable.

Reply:
I think this requires a different fix.  I'm only fixing the symptoms here but not the cause.  The more I look the more issues I see with this download queue.  I'll find a different solution to fix this.  Thanks,

Reply:
Don't worry, proceed with this patch. I'm working on a related issue https://github.com/mozilla-mobile/fenix/issues/9884 and I believe I found the source of the issue.  👍